fhir-bucket/README.md

```diff
@@ -307,3 +312,59 @@ SELECT loader_instance_id, substr(object_name, 1, 24) object_name, resource_type
    AND lr.run_seconds > 0
 ;
 ```
+
+
+## Configuring Logging
+
+Use `-Djava.util.logging.config.file=logging.properties` as the first argument on the command line to configure Java Util Logging using the `logging.properties` file. An example of this file is given below:
+
+```
+handlers=java.util.logging.ConsoleHandler,java.util.logging.FileHandler
+.level=INFO
+
+# Minimal console output
+java.util.logging.ConsoleHandler.level = INFO
+java.util.logging.ConsoleHandler.formatter=com.ibm.fhir.database.utils.common.LogFormatter
+
+# INFO to the log file, unless you want to see more
+java.util.logging.FileHandler.level=INFO
+
+# 50MB * 20 files ~= 1GB of log retention
+java.util.logging.FileHandler.formatter=com.ibm.fhir.database.utils.common.LogFormatter
+java.util.logging.FileHandler.limit=50000000
+java.util.logging.FileHandler.count=20
+java.util.logging.FileHandler.pattern=fhirbucket-%u-%g.log
+
+
+# See FINE stuff for the scanner
+#com.ibm.fhir.bucket.scanner.level=FINE
+```
+
+
+
+## Driving the `$reindex` Custom Operation
+
+When the IBM FHIR Server stores a FHIR resource, it extracts a configurable set of searchable parameter values and stores them in specially indexed tables which are used to support search queries. When the search parameter configuration is changed (perhaps because a profile has been updated), users may want to apply this new configuration to resources already stored. By default, such configuration changes only apply to new resources.
+
+The IBM FHIR Server supports a custom operation to rebuild or "reindex" the search parameters extracted from resources currently stored. The user selects a date or timestamp as the reindex "marker". This value is used to determine which resources have been reindexed, and which still need to be reindexed. When a resource is successfully reindexed, it is marked with this user-selected timestamp. Each reindex REST call will process up to the requested number of resources and return an OperationOutcome resource containing issues describing which resources were processed. When there are no resources left to update, the call returns an OperationOutcome with one issue indicating that the reindex is complete.
+
+To avoid read timeouts, the number of resources processed in a single reindex call can be limited. Reindex calls can be made in parallel to increase throughput. The best number for concurrent requests depends on the capabilities of the underlying platform and any desire to balance load with other users. Concurrency up to 200 threads have been tested. Monitor the IBM FHIR Server response times when increasing concurrency. Also, make sure that the connection pool configured in the FHIR server cluster can support the required number of threads. This also means that the database needs to be configured to support this number of connections (sessions) plus any overhead.
```

Comment:
This feels hidden in fhir-bucket.  Shouldn't it go in fhir-persistence-schema or a Database Design document?

Reply:
fhir-bucket is currently the only practical way to drive to the reindex operation, so seems reasonable to have this here for now. Created issue #1634 1634 to expand the documentation in fhir-persistence-schema.